easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -293,6 +293,24 @@ def prepare_python(self):
         else:
             raise EasyBuildError("Failed to pick Python command to use")
 
+        if self.cfg.get('use_pip'):
+            self.pip_cmd = 'pip'
+            python_ver = get_software_version('Python')
+            if python_ver:
+                if LooseVersion(python_ver) >= LooseVersion('3'):
+                    self.log.info("Python3 detected.  Will use pip3.")
+                    self.install_cmd = self.install_cmd.replace('pip', 'pip3')
+                    self.pip_cmd = 'pip3'
+                    self.log.debug("Setting 'install_cmd' to '%s'", self.install_cmd)
+            else:
+                # Apparently we're using system Python.  Find out version and modify pip command if needed
+                out, _ = run_cmd("%s -V" % (python), simple=False)
```

Comment:
@andreas-h What I had in mind was using `python -V` as a replacement of using `get_software_version`, not a mix of both (less code to maintain is better).

Also, there's a better/easier approach, via `sys.version`:

```
$ python -V
Python 3.6.1
$ python -c "import sys; print(sys.version_info[0])"
3
```

```
$ python -V
Python 2.7.5
$ python -c "import sys; print(sys.version_info[0])"
2
```

That said, I'm starting to have 2nd doubt about this change, it feels like we're working around a problem that shouldn't be there, see also my question at https://lists.ugent.be/wws/arc/easybuild/2017-08/msg00065.html